ambry-server/src/integration-test/java/com.github.ambry.server/RouterServerSSLTest.java

```diff
@@ -35,17 +36,39 @@
 import org.junit.Before;
 import org.junit.BeforeClass;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
 import static com.github.ambry.server.RouterServerTestFramework.*;
 
 
+@RunWith(Parameterized.class)
 public class RouterServerSSLTest {
   private static MockCluster sslCluster;
   private static RouterServerTestFramework testFramework;
   private static MetricRegistry routerMetricRegistry;
   private static long sslSendBytesCountBeforeTest;
   private static long sslReceiveBytesCountBeforeTest;
 
+  /**
+   * Running for both regular and encrypted blobs
+   * @return an array with both {@code false} and {@code true}.
+   */
+  @Parameterized.Parameters
+  public static List<Object[]> data() {
+    return Arrays.asList(new Object[][]{{false}, {true}});
+  }
+
+  /**
+   * Instantiates {@link RouterServerSSLTest}
+   * @param testEncryption {@code true} if blobs need to be tested w/ encryption. {@code false} otherwise
+   */
+  public RouterServerSSLTest(boolean testEncryption) {
+    if (testEncryption) {
+      testFramework.setTestEncryption();
```

Comment:
This method could take in a boolean. This would allow the tests to be executed in any order, instead of encryption staying on once it was enabled.

Reply:
yes. I would want to add it as part of the chain. But due to time constraints, have resorted to this approach. We can work on improving it later.

Reply:
have fixed this based on offline conversation